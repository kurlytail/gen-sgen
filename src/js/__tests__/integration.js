import { execSync } from 'child_process';

describe('# integration test', () => {
    beforeEach(() => {
        execSync('rm -rf testoutput');
    });

    it('## should print help options', () => {
        let output = execSync('npm run build').toString();
        output = execSync('sgen -g `pwd`/dist/sgen.min.js -h').toString();
        expect(output).toMatchSnapshot();
    });

    it('## should generate design and run sgen commands', () => {
        let output = execSync('npm run build').toString();
        output = execSync('sgen -g `pwd`/dist/sgen.min.js -d src/test/fixture/design.json -o testoutput').toString();
        expect(output).toMatchSnapshot();
        output = execSync('npm install', { cwd: 'testoutput' }).toString();
        output = execSync('npm run lint', { cwd: 'testoutput' }).toString();
        output = execSync('npm run build', { cwd: 'testoutput' }).toString();
    });
});
