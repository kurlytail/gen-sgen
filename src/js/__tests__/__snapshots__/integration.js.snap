// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# integration test ## should generate design and run sgen commands 1`] = `
"
info: loading map templates/map.json
info: Opening git repository testoutput/.git
warn: Ignoring error Error: cannot stash changes - there is nothing to stash.
info: Attempting to use branch master-sgen-generated
info: Creating new branch master-sgen-generated
warn: Checkout branch master
warn: Checkout branch master-sgen-generated
info: Generated testoutput/.prettierrc from template templates/.prettierrc._
info: Generated testoutput/.babelrc from template templates/.babelrc._
info: Generated testoutput/.editorconfig from template templates/.editorconfig._
info: Generated testoutput/.eslintignore from template templates/.eslintignore._
info: Generated testoutput/.eslintrc.js from template templates/.eslintrc.js._
info: Generated testoutput/.gitignore from template templates/.gitignore._
info: Generated testoutput/.npmignore from template templates/.npmignore._
info: Generated testoutput/Jenkinsfile from template templates/Jenkinsfile._
info: Generated testoutput/package.json from template templates/package.json._
info: Generated testoutput/README.md from template templates/README.md._
info: Generated testoutput/src/js/__tests__/integration.js from template templates/src/js/__tests__/integration.js._
info: Generated testoutput/src/test/fixture/design.json from template templates/src/test/fixture/design.json._
info: Generated testoutput/templates/map.json from template templates/templates/map.json._
info: Generated testoutput/templates/.prettierrc._ from template templates/templates/.prettierrc._
info: Generated testoutput/.flowconfig from template templates/.flowconfig._
info: Generated testoutput/webpack.config.js from template templates/webpack.config.js._
info: Generated testoutput/src/js/index.js from template templates/src/js/index.js._
info: Generated testoutput/renovate.json from template templates/renovate.json._
info: Generated testoutput/.mergify.yml from template templates/.mergify.yml._
info: Generated testoutput/.npmrc from template templates/.npmrc._
info: Generated testoutput/azure-pipelines.yml from template templates/azure-pipelines.yml._
warn: Checkout branch master-sgen-generated
info: Committed generated files
warn: Checkout branch master
warn: Generator branch master-sgen-generated was modified


info: sgen is done
"
`;

exports[`# integration test ## should print help options 1`] = `
"Usage:
  node sgen [OPTION]

Options:
  -m, --map=ARG+       mapping files
  -d, --design=ARG+    design files
  -e, --extension=ARG+ extension directories or files
  -o, --output=ARG     output directory for generated files
  -g, --generator=ARG+ generators to be used
  -h, --help           display this help
"
`;
